modules/storages/app/components/storages/admin/storage_list_component.rb

```diff
@@ -0,0 +1,118 @@
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+#
+module Storages::Admin
+  class StorageListComponent < ApplicationComponent
+    alias_method :storages, :model
+
+    def call
+      render(BorderBoxComponent.new(padding: :default, scheme: :default)) do |component|
+        header_slot(component)
+        rows_slot(component)
+      end
+    end
+
+    def header_slot(component)
+      component.with_header do |header|
+        header.with_title(tag: :h2) do
+          header_title
+        end
+      end
+    end
+
+    def rows_slot(component)
+      storages.map do |storage|
+        component.with_row(scheme: :default, id: storage_row_css_id(storage)) do
+          storage_row(storage)
+        end
+      end
+    end
+
+    private
+
+    def storage_row(storage)
+      storage_name_div(storage) +
+        div_tag(storage_creator(storage)) +
+        div_tag(storage.provider_type) +
+        div_tag(storage.host)
+    end
+
+    def storage_name_div(storage)
+      div_tag(
+        storage_name(storage) +
+          span_tag("Created on #{storage.created_at.to_fs(:long)}")
+      )
+    end
+
+    def storage_row_css_id(storage)
+      helpers.dom_id storage
+    end
+
+    def header_title
+      helpers.pluralize(storages.size, I18n.t("storages.label_storage"))
+    end
+
+    def storage_name(storage)
+      if storage.configured?
+        span_tag(storage.name)
+      else
+        render(Primer::Beta::Octicon.new(:'alert-fill', size: :small, color: :severe)) +
+          span_tag(storage.name, classes: 'pl-2')
+      end
+    end
+
+    def storage_creator(storage)
+      # TODO: Replace with `Users::AvatarComponent` once https://github.com/opf/openproject/pull/13527 is merged
+      helpers.avatar(storage.creator, size: :mini) +
+        storage.creator.name
+    end
+
+    def span_tag(item, **)
+      base_component_tag(item, tag: :span, **)
+    end
+
+    def div_tag(item, **)
+      base_component_tag(item, tag: :div, **)
+    end
+
+    def base_component_tag(item, tag:, **)
+      render(Primer::BaseComponent.new(tag:, **)) { item }
+    end
+  end
+
+  class BorderBoxComponent < Primer::Beta::BorderBox
+    def before_render
+      super
+
+      # Remove the _base.sass margin-left from <ul> tag
+      @list_arguments[:classes] = "ml-0"
+    end
```

Comment:
Overriding the default list arguments- that are not override able via [the constructor](https://github.com/primer/view_components/blob/main/app/components/primer/beta/border_box.rb#L85) afaics. cc @HDinger

Reply:
This is necessary because `frontend/src/global_styles/openproject/_base.sass` defines this at the end of the file:

```
ul,
ol
  margin-left: 2rem
```

It works fine with a plain `Primer::Beta::BorderBox` if these styles are removed. What would be the side-effects if these lines were removed?

Reply:
The lines in `frontend/src/global_styles/openproject/_base.sass` are necessary for now to intend our list. Without them, all of our list would show their bullet points too far on the left (maybe even out of view). So we need that. I am aware, that inside the Primer::Beta::BorderBox it looks weird if we do not override it.
However, I am reyling on [this PR](https://github.com/opf/openproject/pull/13527) as it will add a general rule for [lists inside BorderBoxes](https://github.com/opf/openproject/pull/13527/files#diff-ebc8270818b7f26cb6ecc2c0c6b4752069453d5b02a91f6f08c5738ac8387c3cR34). We have to wait for that PR anyway to benefit from the flexbox helpers.